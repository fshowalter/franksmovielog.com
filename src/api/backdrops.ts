import { getImage } from "astro:assets";
import path from "node:path";
import sharp from "sharp";

import { normalizeSources } from "./utils/normalizeSources";

export type BackdropImageProps = {
  src: string;
  srcSet: string;
};

const images = import.meta.glob<{ default: ImageMetadata }>(
  "/content/assets/backdrops/*.png",
);

export async function getBackdropImageProps(
  slug: string,
  {
    height,
    width,
  }: {
    height: number;
    width: number;
  },
): Promise<BackdropImageProps> {
  const backdropFile = await getBackdropFile(slug);

  const optimizedImage = await getImage({
    format: "avif",
    height: height,
    quality: 80,
    src: backdropFile.default,
    width: width,
    widths: [0.3, 0.5, 0.8, 1].map((w) => w * width),
  });

  return {
    src: normalizeSources(optimizedImage.src),
    srcSet: normalizeSources(optimizedImage.srcSet.attribute),
  };
}

export async function getOpenGraphBackdropAsBase64String(slug: string) {
  const imageBuffer = await sharp(
    path.resolve(`./content/assets/backdrops/${slug}.png`),
  )
    .resize(1200)
    .toFormat("png")
    .toBuffer();

  return `data:${"image/png"};base64,${imageBuffer.toString("base64")}`;
}

async function getBackdropFile(slug: string) {
  let backdropFilePath = Object.keys(images).find((path) => {
    return path.endsWith(`${slug}.png`);
  })!;

  backdropFilePath =
    backdropFilePath ||
    Object.keys(images).find((path) => {
      return path.endsWith(`default.png`);
    })!;

  return await images[backdropFilePath]();
}
