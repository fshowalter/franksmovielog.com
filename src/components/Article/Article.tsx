import type { BackdropImageProps } from "src/api/backdrops";
import type { MoreReviewsValue } from "src/components/MoreReviews";

import { Backdrop } from "src/components/Backdrop";
import { Layout } from "src/components/Layout";
import { LongFormText } from "src/components/LongFormText";
import { MoreReviews } from "src/components/MoreReviews";
import { SubHeading } from "src/components/SubHeading";

export interface Props {
  backdropImageProps: BackdropImageProps;
  content: null | string;
  deck: string;
  recentReviews: MoreReviewsValue[];
  title: string;
}

export function Article({
  backdropImageProps,
  content,
  deck,
  recentReviews,
  title,
}: Props): JSX.Element {
  return (
    <Layout>
      <article>
        <Backdrop
          deck={deck}
          imageProps={backdropImageProps}
          size="large"
          title={title}
        />
        <section className="flex flex-col items-center pb-32 pt-16">
          <div className="px-container">
            <LongFormText className="max-w-prose" text={content} />
          </div>
        </section>
      </article>
      <div className="bg-subtle pb-16 pt-6">
        <MoreReviews values={recentReviews}>
          <SubHeading as="h2">
            Recent{" "}
            <a className="text-accent" href="/reviews/">
              Reviews
            </a>
          </SubHeading>
        </MoreReviews>
      </div>
    </Layout>
  );
}
