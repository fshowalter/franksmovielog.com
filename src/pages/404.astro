---
import { getOpenGraphBackdropSrc } from "src/api/backdrops";
import { Article } from "src/components/Article/Article";
import { getProps } from "src/components/Article/getProps";
import Layout from "src/layouts/Layout.astro";

const slug = "not-found";

const props = await getProps({
  slug,
  deck: '"Dick Laurent is dead."',
});

const openGraphImageSrc = await getOpenGraphBackdropSrc(slug);
---

<Layout
  title="404: Not Found"
  description="Dick Laurent is dead."
  image={openGraphImageSrc}
  article
>
  <Article {...props} />
</Layout>
