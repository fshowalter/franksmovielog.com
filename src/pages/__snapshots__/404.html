<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width" />
    <link rel="icon" href="/favicon.ico" sizes="48x48" />
    <link rel="icon" href="/favicon.svg" sizes="any" type="image/svg+xml" />
    <link rel="apple-touch-icon" href="/apple-touch-icon.png" />
    <link rel="manifest" href="/site.webmanifest" />
    <link rel="sitemap" href="/sitemap-index.xml" />
    <meta name="description" content="Dick Laurent is dead." />
    <meta
      name="og:image"
      content="/_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&#38;w=1200&#38;h=675&#38;q=80&#38;f=jpeg"
    />
    <meta property="og:url" content="https://www.franksmovielog.com//" />
    <meta property="og:type" content="article" />
    <meta property="og:title" content="404: Not Found" />
    <meta property="og:description" content="Dick Laurent is dead." />
    <title>404: Not Found | Frank&#39;s Movie Log</title>
  </head>
  <body>
    <div>
      <a
        href="#content"
        class="absolute left-1/2 top-0.5 z-50 mx-auto bg-subtle px-6 py-2 text-center text-accent [transform:translate(-50%,calc(-100%_-_2px))] focus:[transform:translate(-50%,0%)]"
        >Skip to content</a
      >
      <div class="flex min-h-full w-full flex-col bg-default">
        <header
          class="z-20 flex w-full items-center justify-between px-container-base py-4 tablet:p-6 desktop:inset-x-0 desktop:z-40 desktop:flex-row desktop:flex-wrap desktop:px-16 desktop:py-8 desktop:text-left"
          style="
            color: #fff;
            position: absolute;
            background-image: linear-gradient(
              to bottom,
              rgba(0, 0, 0, 0.35),
              rgba(0, 0, 0, 0.55) 55%,
              transparent
            );
          "
        >
          <div class="flex flex-col">
            <div
              class="whitespace-nowrap font-normal leading-8"
              style="font-size: 1.5625rem"
            >
              <a href="/">Frank&#x27;s Movie Log</a>
            </div>
            <p class="w-full pl-px text-sm italic leading-4 opacity-85">
              My life at the movies.
            </p>
          </div>
          <div class="flex items-center">
            <nav class="hidden w-full desktop:block desktop:w-auto">
              <ul class="flex flex-wrap justify-start gap-x-6 text-xl">
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/"
                    style="text-shadow: 1px 1px 2px black"
                    >Home</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/how-i-grade/"
                    style="text-shadow: 1px 1px 2px black"
                    >How I Grade</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/reviews/"
                    style="text-shadow: 1px 1px 2px black"
                    >Reviews</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/viewings/"
                    style="text-shadow: 1px 1px 2px black"
                    >Viewing Log</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/cast-and-crew/"
                    style="text-shadow: 1px 1px 2px black"
                    >Cast &amp; Crew</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/collections/"
                    style="text-shadow: 1px 1px 2px black"
                    >Collections</a
                  >
                </li>
                <li class="block whitespace-nowrap tracking-0.5px">
                  <a
                    class="text-inherit transition-all duration-500 ease-in-out hover:bg-[rgba(0,0,0,.50)]"
                    href="/watchlist/"
                    style="text-shadow: 1px 1px 2px black"
                    >Watchlist</a
                  >
                </li>
              </ul>
            </nav>
            <div class="z-[1000]">
              <button
                data-open-modal="true"
                disabled=""
                aria-label="Search"
                aria-keyshortcuts="Control+K"
                class="flex items-center justify-between overflow-hidden text-sm leading-6 ring-default desktop:ml-6"
                type="button"
                title="Search: Control+K"
              >
                <svg
                  xmlns="http://www.w3.org/2000/svg"
                  fill="none"
                  viewBox="0 0 24 24"
                  stroke-width="1.5"
                  stroke="currentColor"
                  class="size-6 desktop:size-7"
                >
                  <path
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
                  ></path>
                </svg>
              </button>
            </div>
            <input type="checkbox" id="mobile-nav" class="hidden" /><label
              for="mobile-nav"
              class="hamburger-icon desktop:hidden"
              ><span style="background: #fff" class="hamburger-icon-bars"></span
            ></label>
            <ul
              class="hamburger-menu flex flex-col items-start gap-y-5 text-left text-inverse desktop:hidden"
            >
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/">Home</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/how-i-grade/">How I Grade</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/reviews/">Reviews</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a href="/reviews/underseen/">Underseen Gems</a>
                  </li>
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a href="/reviews/overrated/">Overrated Disappointments</a>
                  </li>
                </ol>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/viewings/">Viewing Log</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a href="/viewings/stats/">Stats</a>
                  </li>
                </ol>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/cast-and-crew/">Cast &amp; Crew</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/collections/">Collections</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a href="/watchlist/">Watchlist</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a href="/watchlist/progress/">Progress</a>
                  </li>
                </ol>
              </li>
            </ul>
          </div>
        </header>
        <main class="grow" id="content">
          <article>
            <header
              class="relative flex min-h-[240px] content-start items-end bg-cover pb-8 pt-40 text-inverse [background-position-x:center] tablet:min-h-[400px] tablet:pb-10 tablet:pt-40 desktop:min-h-[clamp(640px,50vh,1350px)] desktop:pb-16 desktop:pt-40"
            >
              <img
                class="absolute inset-0 size-full object-cover object-top"
                srcset="
                  /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=600&amp;q=80&amp;f=avif              600w,
                  /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=1200&amp;q=80&amp;f=avif            1200w,
                  /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=2400&amp;h=1350&amp;q=80&amp;f=avif 2400w,
                  /_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=2400&amp;h=1350&amp;q=80&amp;f=avif 4800w
                "
                src="/_image?/content%2Fassets%2Fbackdrops%2Fnot-found.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=2400&amp;h=1350&amp;q=80&amp;f=avif"
                width="2400"
                height="1350"
                loading="eager"
                fetchpriority="high"
                alt="Dick Laurent is dead."
              />
              <div class="z-10 mx-auto w-full max-w-screen-max px-container">
                <h1 class="font-sans-bold text-2xl uppercase desktop:text-7xl">
                  404 Not Found
                </h1>
              </div>
            </header>
            <section class="flex flex-col items-center">
              <div class="spacer-y-16"></div>
              <div class="px-pageMargin">
                <div
                  class="rendered-markdown text-md/7 tracking-0.3px tablet:text-xl/8 max-w-prose"
                >
                  <p>
                    This is probably my fault. Are you looking for one of my old
                    reviews? I’m in the process of touching them up and
                    reposting them. In the mean time, you can find them in the
                    site’s
                    <a href="https://github.com/fshowalter/franksmovielog.com/"
                      >GitHub repo</a
                    >.
                  </p>
                  <p>
                    If you still can’t find what you need,
                    <a
                      href="https://github.com/fshowalter/franksmovielog.com/issues/new/choose"
                      >open an issue on GitHub</a
                    >
                    and I’ll try to help.
                  </p>
                  <p>My apologies, and thanks for stopping by.</p>
                </div>
              </div>
              <div class="spacer-y-32"></div>
            </section>
          </article>
          <section class="bg-subtle pb-16 pt-6">
            <nav data-pagefind-ignore="true">
              <div
                class="relative mx-auto w-full max-w-screen-max px-container desktop:px-20"
              >
                <h2
                  class="py-10 font-sans-bold text-xs uppercase tracking-[0.8px] text-subtle"
                >
                  Recent<!-- -->
                  <a href="/reviews/" class="text-accent">Reviews</a>
                </h2>
                <ul
                  class="flex w-full flex-col gap-x-[2.60416667%] gap-y-6 tablet:flex-row tablet:flex-wrap tablet:justify-between desktop:flex-nowrap"
                >
                  <li class="w-full tablet:w-[47%]">
                    <a href="/reviews/chamber-of-horrors-1966/" class="block"
                      ><img
                        srcset="/_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=108.75&amp;q=80&amp;f=avif 108.75w, /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=217.5&amp;q=80&amp;f=avif 217.5w, /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;q=80&amp;f=avif 435w, /_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=870&amp;q=80&amp;f=avif 870w"
                        src="/_image?/content%2Fassets%2Fstills%2Fchamber-of-horrors-1966.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;h=179&amp;q=80&amp;f=avif"
                        alt="A still from Chamber of Horrors (1966)"
                        loading="lazy"
                        decoding="async"
                        width="435"
                        height="179"
                        sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                        class="aspect-[1.77777778] h-auto w-full"
                    /></a>
                    <div
                      class="flex flex-col bg-default px-6 pb-4 pt-6 desktop:pl-[12%] desktop:pr-[14%]"
                    >
                      <a
                        href="/reviews/chamber-of-horrors-1966/"
                        class="mb-4 block font-serif-semibold text-xl text-default"
                        >Chamber of Horrors<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1966</span
                        ></a
                      ><picture
                        ><source
                          srcset="/svg/2-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/2-stars.svg"
                          alt="D+: 2 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mb-4"
                      /></picture>
                      <div
                        class="rendered-markdown mb-8 text-base leading-[1.6] tracking-0.3px text-muted"
                      >
                        <p>
                          Warner Bros. shot <em>Chamber of Horrors</em> as a
                          television movie, intended as a series pilot. When the
                          network deemed the finished product too violent for
                          television, the studio added some gimmicks and
                          released it as a theatrical feature. The result looks
                          good but underwhelms.
                          <a
                            class="!no-underline uppercase whitespace-nowrap font-normal font-sans-narrow text-accent text-xs tracking-[1.1px] leading-none hover:!underline"
                            href="/reviews/chamber-of-horrors-1966/"
                            >Continue reading...</a
                          >
                        </p>
                      </div>
                      <div
                        class="font-sans-narrow text-xxs uppercase leading-4 tracking-[0.8px] text-subtle"
                      >
                        <span>Crime</span
                        ><span>
                          |
                          <!-- -->Horror</span
                        ><span>
                          |
                          <!-- -->Thriller</span
                        >
                      </div>
                    </div>
                  </li>
                  <li class="w-full tablet:w-[47%]">
                    <a href="/reviews/two-for-the-seesaw-1962/" class="block"
                      ><img
                        srcset="/_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=108.75&amp;q=80&amp;f=avif 108.75w, /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=217.5&amp;q=80&amp;f=avif 217.5w, /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;q=80&amp;f=avif 435w, /_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=870&amp;q=80&amp;f=avif 870w"
                        src="/_image?/content%2Fassets%2Fstills%2Ftwo-for-the-seesaw-1962.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;h=179&amp;q=80&amp;f=avif"
                        alt="A still from Two for the Seesaw (1962)"
                        loading="lazy"
                        decoding="async"
                        width="435"
                        height="179"
                        sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                        class="aspect-[1.77777778] h-auto w-full"
                    /></a>
                    <div
                      class="flex flex-col bg-default px-6 pb-4 pt-6 desktop:pl-[12%] desktop:pr-[14%]"
                    >
                      <a
                        href="/reviews/two-for-the-seesaw-1962/"
                        class="mb-4 block font-serif-semibold text-xl text-default"
                        >Two for the Seesaw<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1962</span
                        ></a
                      ><picture
                        ><source
                          srcset="/svg/3-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/3-stars.svg"
                          alt="C+: 3 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mb-4"
                      /></picture>
                      <div
                        class="rendered-markdown mb-8 text-base leading-[1.6] tracking-0.3px text-muted"
                      >
                        <p>
                          In <em>Two for the Seesaw</em>, Robert Mitchum plays a
                          middle-aged, midwestern lawyer who—after his wife of
                          twelve years asks for a divorce—flees his cushy job
                          and pampered lifestyle for New York City. There, he
                          meets a late-twenty-something dancer played by Shirley
                          MacLaine. A romance blooms.
                          <strong>Spoilers follow</strong>.
                          <a
                            class="!no-underline uppercase whitespace-nowrap font-normal font-sans-narrow text-accent text-xs tracking-[1.1px] leading-none hover:!underline"
                            href="/reviews/two-for-the-seesaw-1962/"
                            >Continue reading...</a
                          >
                        </p>
                      </div>
                      <div
                        class="font-sans-narrow text-xxs uppercase leading-4 tracking-[0.8px] text-subtle"
                      >
                        <span>Drama</span
                        ><span>
                          |
                          <!-- -->Romance</span
                        >
                      </div>
                    </div>
                  </li>
                  <li class="w-full tablet:w-[47%]">
                    <a
                      href="/reviews/the-man-with-nine-lives-1940/"
                      class="block"
                      ><img
                        srcset="/_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=108.75&amp;q=80&amp;f=avif 108.75w, /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=217.5&amp;q=80&amp;f=avif 217.5w, /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;q=80&amp;f=avif 435w, /_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=870&amp;q=80&amp;f=avif 870w"
                        src="/_image?/content%2Fassets%2Fstills%2Fthe-man-with-nine-lives-1940.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;h=179&amp;q=80&amp;f=avif"
                        alt="A still from The Man with Nine Lives (1940)"
                        loading="lazy"
                        decoding="async"
                        width="435"
                        height="179"
                        sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                        class="aspect-[1.77777778] h-auto w-full"
                    /></a>
                    <div
                      class="flex flex-col bg-default px-6 pb-4 pt-6 desktop:pl-[12%] desktop:pr-[14%]"
                    >
                      <a
                        href="/reviews/the-man-with-nine-lives-1940/"
                        class="mb-4 block font-serif-semibold text-xl text-default"
                        >The Man with Nine Lives<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1940</span
                        ></a
                      ><picture
                        ><source
                          srcset="/svg/2-half-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/2-half-stars.svg"
                          alt="C-: 2.5 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mb-4"
                      /></picture>
                      <div
                        class="rendered-markdown mb-8 text-base leading-[1.6] tracking-0.3px text-muted"
                      >
                        <p>
                          Despite Boris Karloff playing a doctor with a secret
                          underground laboratory where he holds six people
                          hostage as unwilling test subjects,
                          <em>The Man with Nine Lives</em> isn’t a mad scientist
                          picture. It’s a drama centered on an ethical dilemma:
                          Would you risk the deaths of six people to save
                          millions?
                          <a
                            class="!no-underline uppercase whitespace-nowrap font-normal font-sans-narrow text-accent text-xs tracking-[1.1px] leading-none hover:!underline"
                            href="/reviews/the-man-with-nine-lives-1940/"
                            >Continue reading...</a
                          >
                        </p>
                      </div>
                      <div
                        class="font-sans-narrow text-xxs uppercase leading-4 tracking-[0.8px] text-subtle"
                      >
                        <span>Horror</span
                        ><span>
                          |
                          <!-- -->Mystery</span
                        ><span>
                          |
                          <!-- -->Sci-Fi</span
                        >
                      </div>
                    </div>
                  </li>
                  <li class="w-full tablet:w-[47%]">
                    <a
                      href="/reviews/burial-ground-the-nights-of-terror-1981/"
                      class="block"
                      ><img
                        srcset="/_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=108.75&amp;q=80&amp;f=avif 108.75w, /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=217.5&amp;q=80&amp;f=avif 217.5w, /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;q=80&amp;f=avif 435w, /_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=870&amp;q=80&amp;f=avif 870w"
                        src="/_image?/content%2Fassets%2Fstills%2Fburial-ground-the-nights-of-terror-1981.png%3ForigWidth%3D2000%26origHeight%3D1125%26origFormat%3Dpng&amp;w=435&amp;h=179&amp;q=80&amp;f=avif"
                        alt="A still from Burial Ground: The Nights of Terror (1981)"
                        loading="lazy"
                        decoding="async"
                        width="435"
                        height="179"
                        sizes="(min-width: 706px) 312px, (min-width: 1280) 25vw, (min-width: 1472px) 312px, 50vw"
                        class="aspect-[1.77777778] h-auto w-full"
                    /></a>
                    <div
                      class="flex flex-col bg-default px-6 pb-4 pt-6 desktop:pl-[12%] desktop:pr-[14%]"
                    >
                      <a
                        href="/reviews/burial-ground-the-nights-of-terror-1981/"
                        class="mb-4 block font-serif-semibold text-xl text-default"
                        >Burial Ground: The Nights of Terror<!-- -->
                        <span class="text-sm font-light leading-none text-muted"
                          >1981</span
                        ></a
                      ><picture
                        ><source
                          srcset="/svg/3-stars-dark.svg"
                          media="(prefers-color-scheme: dark)" />
                        <img
                          src="/svg/3-stars.svg"
                          alt="C: 3 stars (out of 5)"
                          height="18"
                          width="90"
                          class="mb-4"
                      /></picture>
                      <div
                        class="rendered-markdown mb-8 text-base leading-[1.6] tracking-0.3px text-muted"
                      >
                        <p>
                          Teamwork wins. That’s the takeaway from
                          <em>Burial Ground</em>, an Italian zombie horror that
                          packs just enough originality to set itself above the
                          mass of <em>Zombie</em> knock-offs.
                          <a
                            class="!no-underline uppercase whitespace-nowrap font-normal font-sans-narrow text-accent text-xs tracking-[1.1px] leading-none hover:!underline"
                            href="/reviews/burial-ground-the-nights-of-terror-1981/"
                            >Continue reading...</a
                          >
                        </p>
                      </div>
                      <div
                        class="font-sans-narrow text-xxs uppercase leading-4 tracking-[0.8px] text-subtle"
                      >
                        <span>Horror</span>
                      </div>
                    </div>
                  </li>
                </ul>
              </div>
            </nav>
          </section>
        </main>
        <footer
          class="flex flex-wrap items-start justify-between gap-[10%] bg-[#252525] px-container-base py-20 text-inverse tablet:px-12 tablet:pt-10 desktop:p-20"
        >
          <div class="flex flex-col">
            <div class="flex flex-col">
              <div
                class="whitespace-nowrap font-normal leading-8"
                style="font-size: 1.5625rem"
              >
                <a href="/">Frank&#x27;s Movie Log</a>
              </div>
              <p class="w-full pl-px text-sm italic leading-4 opacity-85">
                My life at the movies.
              </p>
            </div>
            <div class="spacer-y-12"></div>
          </div>
          <div
            class="flex flex-col gap-20 pb-20 pt-10 tablet:basis-1/2 tablet:pt-20"
          >
            <ul class="flex w-full flex-col gap-y-10 text-inverse max:w-auto">
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/">Home</a>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/how-i-grade/"
                  >How I Grade</a
                >
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/reviews/">Reviews</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a class="hover:text-inverse" href="/reviews/underseen/"
                      >Underseen Gems</a
                    >
                  </li>
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a class="hover:text-inverse" href="/reviews/overrated/"
                      >Overrated Disappointments</a
                    >
                  </li>
                </ol>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/viewings/">Viewing Log</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a class="hover:text-inverse" href="/viewings/stats/"
                      >Stats</a
                    >
                  </li>
                </ol>
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/cast-and-crew/"
                  >Cast &amp; Crew</a
                >
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/collections/"
                  >Collections</a
                >
              </li>
              <li class="block w-1/2 whitespace-nowrap text-2xl">
                <a class="hover:text-accent" href="/watchlist/">Watchlist</a>
                <ol class="mt-4">
                  <li
                    class="mb-2 font-sans-narrow text-sm uppercase tracking-[1px] text-subtle"
                  >
                    <a class="hover:text-inverse" href="/watchlist/progress/"
                      >Progress</a
                    >
                  </li>
                </ol>
              </li>
            </ul>
            <a
              href="#top"
              class="mb-8 w-full max-w-[320px] bg-canvas px-pageMargin py-5 text-center font-sans-narrow text-sm uppercase tracking-[.6px] text-default hover:bg-inverse hover:text-inverse"
              >To the top</a
            >
            <p class="w-full font-light leading-4 text-subtle">
              All stills used in accordance with the<!-- -->
              <a
                href="http://www.copyright.gov/title17/92chap1.html#107"
                class="text-inherit"
                >Fair Use Law.</a
              >
            </p>
          </div>
        </footer>
      </div>
    </div>
    <dialog
      class="relative m-0 h-screen max-h-none w-screen max-w-unset border border-default bg-subtle shadow-lg open:flex open:flex-col tablet:mx-auto tablet:mt-16 tablet:h-min tablet:max-h-[min(100vh_-_128px,900px)] tablet:min-h-60 tablet:max-w-[min(100vw_-_2_*_var(--gutter-width),36rem)] tablet:rounded-xl desktop:min-h-0"
      aria-label="Search"
    >
      <div data-dialog-frame class="contents">
        <button
          data-close-modal
          class="px-4 font-sans-narrow text-sm hover:bg-inverse hover:text-inverse"
        >
          Cancel
        </button>
        <div class="contents"><div id="pagefind__search"></div></div>
        <div class="hidden px-container py-4 desktop:block">
          <kbd class="rounded-md bg-default p-2 font-mono text-xs text-subtle"
            >Tab</kbd
          >
          to navigate
          <kbd class="rounded-md bg-subtle p-2 font-mono text-xs text-subtle"
            >ESC</kbd
          >
          to close
        </div>
      </div>
    </dialog>
    <script>
      (() => {
        const openBtn = document.querySelector("button[data-open-modal]");
        if (!openBtn) return;
        if (/(Mac|iPhone|iPod|iPad)/i.test(navigator.platform)) {
          openBtn.setAttribute("aria-keyshortcuts", "Meta+K");
          openBtn.setAttribute("title", `Search: ⌘K`);
        }
      })();
    </script>
  </body>
</html>
